pyiron_atomistics/atomistics/structure/factory.py

```diff
@@ -407,3 +416,105 @@ def from_pymatgen(pymatgen_obj):
+            For the given choice I could not find any symmetrically equivalent vector that lies in the terrace.\
+            please change the kinkOrientation and try again')
+        temp = (np.cross(np.squeeze(eqvdirk[eqvdirk_ind[0]]), np.squeeze(eqvdirs))).tolist().index(terrace_orientation)
+        fin_kink_orientation = eqvdirk[eqvdirk_ind[0]]
+        fin_step_orientation = eqvdirs[temp]
+        vec1 = (np.asanyarray(fin_step_orientation).dot(length_step)) + \
+               (np.asanyarray(fin_kink_orientation).dot(length_kink))
+        vec2 = (np.asanyarray(fin_kink_orientation).dot(length_terrace)) + step_down_vector
+        high_index_surface = np.cross(np.asanyarray(vec1), np.asanyarray(vec2))
+        high_index_surface = np.array(high_index_surface / np.gcd.reduce(high_index_surface), dtype=int)
+
+        return high_index_surface, fin_kink_orientation, fin_step_orientation
+
+    def high_index_surface(self, element='Ni', crystal_structure='fcc', lattice_constant=3.526,
+                           terrace_orientation=None, step_orientation=None, kink_orientation=None,
+                           step_down_vector=None, length_step=3, length_terrace=3, length_kink=1, layers=60,
+                           vacuum=10):
+        """
+        Gives a slab positioned at the bottom with the high index surface computed by high_index_surface_info().
+        Args:
+            element (str): The parent element eq. "N", "O", "Mg" etc.
+            crystal_structure (str): The crystal structure of the lattice
+            lattice_constant (float): The lattice constant
+            terrace_orientation (list): The miller index of the terrace. default: [1,1,1]
+            step_orientation (list): The miller index of the step. default: [1,1,0]
+            kink_orientation (list): The miller index of the kink. default: [1,1,1]
+            step_down_vector (list): The direction for stepping down from the step to next terrace. default: [1,1,0]
+            length_terrace (int): The length of the terrace along the kink direction in atoms. default: 3
+            length_step (int): The length of the step along the step direction in atoms. default: 3
+            length_kink (int): The length of the kink along the kink direction in atoms. default: 1
+            layers (int): Number of layers of the high_index_surface. default: 60
+            vacuum (float): Thickness of vacuum on the top of the slab. default:10
+
+        Returns:
+            slab: pyiron_atomistics.atomistics.structure.atoms.Atoms instance Required surface
+        """
+        basis = self.crystal(element=element, bravais_basis=crystal_structure, lattice_constant=lattice_constant)
+        high_index_surface, \
+            fin_kink_orientation, \
+            fin_step_orientation = self.high_index_surface_info(element='Ni',
+                                                                crystal_structure='fcc', lattice_constant=3.526,
```

Comment:
Here, you need to pass the given arguments of the function call! 
```suggestion
            fin_step_orientation = self.high_index_surface_info(element=element,
                                                                crystal_structure=crystal_structure, lattice_constant=lattice_constant,
```
Indeed, I would prefer no defaults for these in the first place.